### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by BahirHakimy**
# In this branch I have done the following
- Created `Person`, `Student`, `Teacher` classes
- Added every class to its own file
- Setup Linters
- Used Gitflow

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def can_use_services?
+    return true if of_age? || @parent_permission
+
+    false
+  end
```
Comment (ridwanediallo) [inline person.rb]: - [OPTIONAL ] To make sure that the can_use_services? looks more straightforward and simple I would recommend you refactor it to this :point_down: 

```
  def can_use_services?
      of_age? || @parent_permission
  end

```

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  private
+
+  def of_age?
+    return true if @age >= 18
+
+    false
+  end
```
Comment (ridwanediallo) [inline person.rb]: - [OPTIONAL ] I would recommend refactoring the of_age? method to this to make it look more straightforward and simple. Like this :point_down: 

```
  def of_age?
      @age >= 18
  end
```

**Review by ridwanediallo — APPROVED** [review]
# STATUS: APPROVED 🟢
## Hi @BahirHakimy , 
Congratulations!GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used GitHub flow correctly. ✔
- Your PR looks professionally. ✔

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 2 above)
Comment (BahirHakimy) [reply person.rb]: Thank you very much @ridwanediallo for approving my pull request and giving insightful suggestions, I will definitely implement them 🍻

**Pull request merged**